crates/cadenza-eval/src/ir/mod.rs

```diff
@@ -0,0 +1,490 @@
+//! Intermediate Representation (IR) for Cadenza
```

Comment:
maybe let's just focus on the design documents in this PR and then we can get the design right before implementing it. so let's delete the `ir` modules for now.